argocd/structure_application.go

```diff
@@ -311,6 +311,10 @@ func expandApplicationSourceHelm(in []interface{}) *application.ApplicationSourc
 		result.SkipCrds = v.(bool)
 	}
 
+	if v, ok := a["version"]; ok {
```

Comment:
This code ensures that the version is correctly set when creating the application in ArgoCD however, you are missing the oppositye side of this code which ensures that the Terraform state is updated to reflect the current state of the application in ArgoCD. See [flattenApplicationSourceHelm](https://github.com/argoproj-labs/terraform-provider-argocd/blob/dfa0f72848fc1bb90f6ffe8cc9f943dd0f7c92e6/argocd/structure_application.go#L762).